sdk/metricsadvisor/ai-metrics-advisor/src/metricsAdvisorKeyCredentialPolicy.ts

```diff
@@ -18,13 +18,70 @@ const X_API_KEY_HEADER_NAME = "x-api-key";
  * Credential used to authenticate and authorize with Metrics Advisor service
  */
 export class MetricsAdvisorKeyCredential {
+  /**
+   * Subscription access key from the Azure portal
+   */
+  private subscriptionKey: string;
+
+  /**
+   * API key from the Metrics Advisor web portal
+   */
+  private apiKey: string;
+
   /**
    * Creates an instance of MetricsAdvisorKeyCredential
    *
    * @param subscriptionKey - Subscription access key from the Azure portal
    * @param apiKey - API key from the Metrics Advisor web portal
    */
-  constructor(readonly subscriptionKey: string, readonly apiKey: string) {}
+  constructor(subscriptionKey: string, apiKey: string) {
+    this.subscriptionKey = subscriptionKey;
```

Comment:
please validate the parameter values here too.

Reply:
```suggestion
    if(!subscription && !apiKey)
    this.subscriptionKey = subscriptionKey;
```